Tree_Preorder_Traversal/Tree_Preorder_Traversal.js.js

```diff
@@ -0,0 +1,98 @@
+
+//Tree_Preorder_Traversal in javascript
+class Node
+{
+    constructor(data)
+    {
+        this.data = data;
+        this.left = null; //left child of the node
+        this.right = null; //right child of the node
+    }
+}
+
+class BinarySearchTree
+{
+    constructor()
+    {
+        this.root = null;
+    }
+//data inserting in tree
+
+    insert(data)
+    {
+        let node = new Node(data);
+        if(this.root == null)
+        {
+            this.root = node;
+        }
+        else
+        {
+            this.insertNode(this.root, node);
+        }
+    }
+
+    insertNode(root, newNode)
+    {
+        if(newNode.data < root.data)
+        {
+            if(root.left == null)
+            {
+                root.left = newNode;
+            }
+            else
+            {
+                this.insertNode(root.left, newNode);
+            }
+        }
+        else if(newNode.data > root.data)
+        {
+            if(root.right == null)
+            {
+                root.right = newNode;
+            }
+            else
+            {
+                this.insertNode(root.right, newNode); 
+            }
+        }
+    }
+
+    getRootNode()
+    {
+        return this.root;
+    }
+//function of preorder traversal
```

Comment:
Please add 4 space indentation to this comment also.

Reply:
Done mam